### ISSUE TITLE
Datepicker 12 hours AM PM 

### ISSUE THREAD

**Issue by FurkanKalaycioglu** [ant-design/ant-design#50132]
### Reproduction link

[![Edit on StackBlitz](https://developer.stackblitz.com/img/open_in_stackblitz.svg)](https://stackblitz.com/edit/antdesigndatepicker5-19-3)

### Steps to reproduce

Not sure whether luxon dates are important here, i just meant to show with correct versions (antd 5.14) the provided link for reproduction works.

### What is expected?

AM PM values should be listed correctly

### What is actually happening?

A
A

is shown

| Environment | Info |
| --- | --- |
| antd | 5.19.3 |
| React | 18.2 |
| System | macOS 14.2 |
| Browser | All browsers |

---
Not sure whether luxon dates are important here, i just meant to show with correct versions (antd 5.14) the provided link for reproduction works.

**Comment by Tarkik3798** [comment]
![Screenshot from 2024-08-08 12-25-29](https://github.com/user-attachments/assets/a057cee0-9bb9-4814-a79c-fa7bfed7429c)
It seems it is there already with 12 Hours format. Can you please double check on your side? or give more context about the issue.

**Comment by FurkanKalaycioglu** [comment]
@Tarkik3798 Yes there's already 12 hour formats. Have you checked the stackblitz, using formats with Luxon at previous versions AM PM were listed correctly. With the new versions and same code you can see there's something wrong with AM PM Selection
![image](https://github.com/user-attachments/assets/3fd17309-2d18-45d4-8f7d-974f1431d2e5)


you can try downgrading the antdesign version on the provided reproduction and it will work correctly.

**Comment by kiner-tang** [comment]
@FurkanKalaycioglu It seems that the latest antd version is working as expected, you can try upgrading the latest version to give it a try
![image](https://github.com/user-attachments/assets/1d2182b3-d692-45cc-88a8-43d8e6f8b1f0)
![image](https://github.com/user-attachments/assets/4c8fc610-2d98-4299-bfd5-1e81cda9fcb0)

**Comment by afc163** [comment]
Reviewed against the latest `master` (`0b1021909f`). The 12-hour picker no longer renders both meridiem options as the same initial.

A maintainer retested the latest version with the reported setup and showed distinct AM and PM values; no later comment contradicted that current-version result.

Closing this issue accordingly. If a minimal reproduction still demonstrates different behavior on the current `master`, please open a new issue with that reproduction.
